### ISSUE TITLE
log/slog: invalid example for slog.Group

### ISSUE THREAD

**Issue by firefart** [golang/go#61885]
The example at https://pkg.go.dev/log/slog#Group does not compile as it uses an internal package ("log/slog/internal/slogtest").

```text
Output:

package play
	prog.go:5:2: use of internal package log/slog/internal/slogtest not allowed
```

The example is located here:
https://github.com/golang/go/blob/master/src/log/slog/example_test.go

It also looks like the examples allow the use of internal packages which might be worth a more general design discussion.

**Comment by seankhliao** [comment]
cc @jba

**Comment by jba** [comment]
This was a deliberate decision: we preferred factoring out common code to making the examples playable. We could revisit.

/cc @adonovan

**Comment by henvic** [comment]
I also agree that revisiting it would be a nice idea as a lot of people learn by using examples as references, and this might be confusing to newcomers.

**Comment by AlekSi** [comment]
That one seems to be resolved.

**Comment by seankhliao** [comment]
done in [CL 536215](https://go.dev/cl/536215)
